src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/FrozenHashTable.cs

```diff
@@ -51,7 +53,7 @@ public static FrozenHashTable Create<T>(T[] entries, Func<T, int> hasher, Action
                 hashCodes[i] = hasher(entries[i]);
             }
 
-            int numBuckets = CalcNumBuckets(hashCodes);
+            int numBuckets = CalcNumBuckets(hashCodes, optimizeForReading);
             ulong fastModMultiplier = HashHelpers.GetFastModMultiplier((uint)numBuckets);
 
             var chainBuddies = new Dictionary<uint, List<ChainBuddy>>();
```

Comment:
What's a "ChainBuddy"? Choice of naming looks kind of off.

Reply:
Yeah, I didn't clean that up, but we probably should at some point.  I believe Martin is just describing all of the items that are in the same hash bucket, and this little struct just stores the item's hashcode and its index in the original array. "chain" as in list of things, and "buddies" as in friends hanging out together.